tests/test/opaque_types.rs

```diff
@@ -46,6 +46,26 @@ fn opaque_reveal() {
     }
 }
 
+#[test]
+fn opaque_where_clause() {
+    test! {
+        program {
+            struct Ty { }
+
+            trait Clone { }
+            trait Trait { }
+            impl Trait for Ty { }
+            opaque type T: Clone where T: Trait = Ty;
+        }
+
+        goal {
+            T: Trait
```

Comment:
This test in particular doesn't seem right. The real role of where-clauses is to inform the "well-formedness conditions" for an opaque type, not to allow us to prove things about the placeholder type. This is the role of the bounds.